### DISCUSSION TITLE
How do I escape single quotes ( ' )?

### DISCUSSION THREAD

**Discussion by Laser411** [mkdocs/mkdocs#3133] [Q&A] [answered]
I need to display a apostrophe I.E "A hero's journey" in the title of a nav section as well as the directory path. How do I display that without throwing code errors? I tried the yaml escape "\" but that did not work.

**Comment by kamilkrzyskow** [comment, marked as answer]
This nav works:
```yaml
nav:
  - "A hero's journey":
    - "A hero's journey/index.md"
```
and this one too:
```yaml
nav:
  - "A hero's journey": "A hero's journey/index.md"
```
and this is the link after copying it:
`http://127.0.0.1:8000/A%20hero%27s%20journey/` 
it's URL encoded, your browser typically decodes it to a better looking `http://127.0.0.1:8000/A hero's journey/`

**Reply by Andre601** [reply]
That's one reason why you shouldn't use spaces and non-alphanummeric characters in folder and file names, as they will be encoded. And while browser may render them is it not a guarantee.

Also, you can use `''` (two single quotes) to escape a quote inside other quotes.

Example: `'Someone''s text'`

**Reply by kamilkrzyskow** [reply]
> That's one reason why you shouldn't use spaces and non-alphanummeric characters in folder and file names, as they will be encoded. And while browser may render them is it not a guarantee.

A lot of sites use non-Latin characters for example Japanese ones, and percent encoding is a standard since ages, as far as I know there is nothing that replaces it. All modern browsers will support it, but whether or not they render them is rather custom 🤷 . I do agree that sticking to normal Latin lowercase characters is ideal, because it simplifies the URL lookup immensely.

> Also, you can use `''` (two single quotes) to escape a quote inside other quotes.
> 
> Example: `'Someone''s text'`

I didn't know about that 🤔 It looks to me like 2 separate strings to me, I confirm it worked, but personally wouldn't use it. Don't know if it's a YAML or MkDocs YAML addition.

**Reply by Andre601** [reply]
Default yaml behaviour.
MkDocs to my knowledge doesn't have any special yaml parsing applied here.

**Reply by Laser411** [reply]
Thank you, using double quotes did indeed fix it, weird that it doesn't have just an escape or something but I will take it. Thank you for your help :D
